File: 기타/201031 프로그래머스 챌린지/1번.py
def solution(n, delivery):
    answer = ['?']*n
    delivery = sorted(delivery, key=lambda x: -x[2])

    for i in delivery:
        if i[2] == 1:
            answer[i[0]-1] = 'O'
            answer[i[1]-1] = 'O'
        else:
            if answer[i[0]-1] == 'O':
                answer[i[1]-1] = 'X'
            elif answer[i[1]-1] == 'O':
                answer[i[0]-1] = 'X'

    return "".join(answer)


delivery = [[5, 6, 0], [1, 3, 1], [1, 5, 0], [7, 6, 0], [3, 7, 1], [2, 5, 0]]
n = 7
print(solution(n, delivery))

File: 기타/210403 카카오/1번/1번.js
function solution(gift_cards, wants) {
  let answer = 0;

  for (let i = 0; i < gift_cards.length; i++) {
    for (let j = i + 1; j < gift_cards.length; j++) {
      if (gift_cards[j] === wants[i]) {
        [gift_cards[i], gift_cards[j]] = [gift_cards[j], gift_cards[i]];
        break;
      }
    }
  }

  // 앞에서부터 뒤에 번호들 중 내 wants와 일치하는게 있으면 스왑

  for (let i = 0; i < gift_cards.length; i++) {
    if (gift_cards[i] !== wants[i]) {
      answer++;
    }
  }

  // 다른 개수 세기

  return answer;
}

const gc = [4, 5, 3, 2, 1];
const wa = [2, 4, 4, 5, 1];

console.log(solution(gc, wa));
